Dockerfile

```diff
@@ -20,115 +50,100 @@ RUN apt-get -y update \
+
+RUN apt-get build-dep -y gridengine && apt-get update -y
 
 # Install Connectome Workbench
 RUN apt-get update && apt-get -y install connectome-workbench=1.2.3-1~nd14.04+1
 
-ENV CARET7DIR=/usr/bin
 
-# Install HCP Pipelines and MSM binaries
-WORKDIR /opt
-RUN apt-get -y update \
+# Install HCP Pipelines
+RUN apt-get update \
     && apt-get install -y --no-install-recommends python-numpy && \
-    wget -q https://github.com/Washington-University/Pipelines/archive/v3.17.0.tar.gz -O pipelines.tar.gz && \
-    tar zxf pipelines.tar.gz && \
-    mv /opt/Pipelines-* /opt/HCP-Pipelines && \
-    rm pipelines.tar.gz && \
-    wget -q https://www.doc.ic.ac.uk/~ecr05/MSM_HOCR_v1/MSM_HOCR_v1-download.tgz -O MSMs.tar.gz && \
-    tar zxf MSMs.tar.gz && \
-    rm MSMs.tar.gz && \
-    mv MSM_HOCR_v1/Ubuntu /opt/HCP-Pipelines/MSMBinaries && \
-    rm -rf MSM_HOCR_v1
-
-WORKDIR /
-
+    wget https://github.com/Washington-University/Pipelines/archive/v3.17.0.tar.gz -O pipelines.tar.gz && \
+    cd /opt/ && \
+    mkdir /opt/HCP-Pipelines && \
+    tar zxf /pipelines.tar.gz -C /opt/HCP-Pipelines --strip-components=1 && \
+    rm /pipelines.tar.gz
+
+# Install FIX
+RUN apt-get update && apt-get install -y build-essential libpcre3 libpcre3-dev  fort77 xorg-dev libbz2-dev liblzma-dev libblas-dev gfortran gcc-multilib gobjc++ libreadline-dev bzip2 libcurl4-gnutls-dev default-jdk gdebi
+
+RUN cd /opt && \
+    wget http://www.fmrib.ox.ac.uk/~steve/ftp/fix.tar.gz && \
+    tar zxvf fix.tar.gz && \
+    rm fix.tar.gz
+RUN mv /opt/fix* /opt/fix
+
+RUN cd /opt && \
+    wget https://cloud.r-project.org/bin/linux/ubuntu/trusty/r-base-core_3.4.4-1trusty0_amd64.deb && \
+    wget https://cloud.r-project.org/bin/linux/ubuntu/trusty/r-base-dev_3.4.4-1trusty0_all.deb && \
+    gdebi -n r-base-core_3.4.4-1trusty0_amd64.deb && \
+    gdebi -n r-base-dev_3.4.4-1trusty0_all.deb && \
+    apt-get install -y libssl-dev 
+
+RUN R --vanilla -e "install.packages('coin', repos='http://cran.us.r-project.org', dependencies=TRUE)" -e "install.packages('strucchange', repos='http://cran.us.r-project.org', dependencies=TRUE)" -e "install.packages('https://cran.r-project.org/src/contrib/Archive/party/party_1.0-25.tar.gz', repos=NULL, type='source')" -e "install.packages('https://cran.r-project.org/src/contrib/Archive/kernlab/kernlab_0.9-24.tar.gz', repos=NULL, type='source')" -e "install.packages('ROCR', repos='http://cran.us.r-project.org', dependencies=TRUE)" -e "install.packages('https://cran.r-project.org/src/contrib/Archive/e1071/e1071_1.6-7.tar.gz', repos=NULL, type='source')" -e "install.packages('https://cran.r-project.org/src/contrib/Archive/randomForest/randomForest_4.6-12.tar.gz', repos=NULL, type='source')"
+
+RUN mkdir /tmp/v83 && \
+    cp /opt/fix*/compiled/Linux/x86_64/MCRInstaller.zip /tmp/v83 && \
+    cd /tmp/v83 && \
+    unzip MCRInstaller.zip
+
+COPY modified_files/MCR_installer_input_v83.txt /tmp/v83/MCR_installer_input.txt
```

Comment:
Could you elaborate on the changes required to this file? It might be clearer to apply them via `diff` instead of replacing a file.